packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -333,9 +333,9 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
       testTimeRecorder = TestTimeRecorder(globals.logger);
     }
 
-    if (buildInfo.packageConfig['test_api'] == null) {
+    if (buildInfo.packageConfig['flutter_test'] == null) {
```

Comment:
so the fix here is to depend on `flutter_test`, which is not published to pub, and thus must be obtained via source from the flutter sdk, and thus guaranteed to depend on the exact same version of `test_api` as the tool, is that right?

Reply:
Yes, that is how I understand it. The problems we had before was if the app developer had test dependencies (`package:test_api`, `package:test`) in their pubspec that had version conflicts with what was in the flutter sdk.

Reply:
Yes exactly, the `flutter test` command itself is compiled with a version of `test_api` that is from the flutter SDK. The user needs to be on exactly that same version in their local package to ensure compatibility.